libs/core/langchain_core/schema/document_loader.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Iterator, List, Optional
+
+from langchain_core.documents import Document
+from langchain_core.schema.text_splitter import TextSplitterInterface
+
+
+class DocumentLoaderInterface(ABC):
+    """Interface for Document Loader.
+
+    Implementations should implement the lazy-loading method using generators
+    to avoid loading all Documents into memory at once.
+
+    The `load` method will remain as is for backwards compatibility, but its
+    implementation should be just `list(self.lazy_load())`.
+    """
+
+    @abstractmethod
+    def load(self) -> List[Document]:
+        """Load data into Document objects.
+
+        Sub-classes should implement this method
+        as `return list(self.lazy_load())`
+        This method returns a List which is materialized in memory.
+        """
+
+    @abstractmethod
+    def load_and_split(
```

Comment:
We need to figure out how to deprecate load_and_split

Reply:
Yeah, It is definitely a helper function